addons/website_sale_product_configurator/views/templates.xml

```diff
@@ -30,10 +30,10 @@
         <table class="table table-striped table-sm">
         <thead>
             <tr>
-                <th class="td-img">
```

Comment:
I wouldn't move this class (nor change the template in general since you're modifying a stable version). Could you instead change the css selector? Something like:
```css
.oe_advanced_configurator_modal th.td-img + th
```
I agree this might not be the most elegant fix (ideally I'd add a new class on the element you want to hide) but it's better not to modify the template.